Jenkinsfile.deploy

```diff
@@ -28,104 +41,86 @@ pipeline {
+        // Here under is a list of env variable which are defined dynamically into the pipeline
+        // We don't define them here because then there would be immutable ...
 
+        // Deploy image information
         // DEPLOY_IMAGE_REGISTRY_CREDENTIALS
         // DEPLOY_IMAGE_USE_OPENSHIFT
         // DEPLOY_IMAGE_REGISTRY
         // DEPLOY_IMAGE_NAMESPACE
         // DEPLOY_IMAGE_NAME_SUFFIX
         // DEPLOY_IMAGE_TAG
-
+        
         // TEST_PARAMS => parameters which will be set in `configure testing` stage, depending also on pipeline params
     }
     
     stages {
-        stage('Clean Workspace') {
-            steps {
-                dir ("${WORKING_DIR}") {
-                    deleteDir()
-                }
-            }
-        }
         stage('Initialize') {
             steps {
                 script {
-                    sh "mkdir -p ${WORKING_DIR} && cd ${WORKSPACE} && cp -Rap * ${WORKING_DIR}"
-                    sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+                    // Make sure Openshift is available and can authenticate before continuing
+                    loginOpenshift()
                     
                     setupImageEnvVarsFromParams("","DEPLOY")
 
                     if (params.DISPLAY_NAME != "") {
                         currentBuild.displayName = params.DISPLAY_NAME
                     }
 
-                    if (env.BRANCH_NAME != "") {
-                        // Switch to branch if not on a multibranch pipeline
-                        env.BRANCH_NAME = params.BUILD_BRANCH_NAME
-                        dir("${WORKING_DIR}"){
-                            checkout([$class: 'GitSCM', branches: [[name: env.BRANCH_NAME]], browser: [$class: 'GithubWeb', repoUrl: "${GIT_URL}"], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: '']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'user-kie-ci10', url: "${GIT_URL}"]]])
-                        }
-                    }
+                    checkout(githubscm.resolveRepository("kogito-cloud-operator", params.GIT_AUTHOR, params.BUILD_BRANCH_NAME, false))
                 }
             }
         }
         stage('Build Kogito Operator') {
             steps {
-                dir ("${WORKING_DIR}") {
-                    sh """
-                        export GOROOT=`go env GOROOT`
-                        go get -u golang.org/x/lint/golint
-                        touch /etc/sub{u,g}id
-                        usermod --add-subuids 10000-75535 \$(whoami)
-                        usermod --add-subgids 10000-75535 \$(whoami)
-                        cat /etc/subuid
-                        cat /etc/subgid
-                        make image_builder=buildah
-                    """
-                }
+                sh """
+                    go get -u golang.org/x/lint/golint
```

Comment:
is the rhba env running on a container?
if not probably there is no need to do this step everytime. unless the directory the dependencies are saved is cleaned up after the build.

Reply:
The jobs run on VM which is destroyed after the job is complete, for every PR a new VM is created, so yes it'll be a clean run